### PR TITLE
USB Reattach fixes + Specify reset time remaining + secrets != 0xFF

### PR DESCRIPTION
**Pull request by tyeth**
This is desired with the pico2w release (with #664 ) to improve user first run experience

### CODE REVIEW

File: src/Wippersnapper.cpp [hunk 1]
```diff
@@ -2500,8 +2500,10 @@ void Wippersnapper::runNetFSM() {
 */
 /**************************************************************************/
 void Wippersnapper::haltError(String error, ws_led_status_t ledStatusColor) {
-  for (;;) {
-    WS_DEBUG_PRINT("ERROR [WDT RESET]: ");
+  for (int i = 0;; i++) {
```
Comment (brentru) [inline src/Wippersnapper.cpp:2503]: This is a really clever count-up system to dynamically feed the wdt!

Before the loop, I would like to see these magic numbers defined as const ints: `25`, `20, `5000`, to help future maintainers read through this code

**Review by brentru — CHANGES_REQUESTED** [review]
This addresses the TinyUSB reattach issue we've been seeing and I like the count-up WDT feed function.

One small comment for code clarity:

**Comment by tyeth** [comment]
@brentru this one updates haltError to use an optional time argument, and parameterises the timeouts

**Review by brentru — APPROVED** [review]
Much cleaner! LGTM!

**Pull request merged**